### PR TITLE
Fix incorrect "VALID UNTIL" assumption made for roles in node activation

### PR DESCRIPTION
**Pull request by onurctirtir**
Fixes https://github.com/citusdata/citus/issues/7533.

DESCRIPTION: Fixes incorrect `VALID UNTIL` setting assumption made for roles when syncing them to new nodes

### CODE REVIEW

**Comment by onurctirtir** [comment]
Upgrade tests are still failing --need to take a closer look.

File: src/test/regress/expected/seclabel.out [hunk 1]
```diff
@@ -167,9 +167,9 @@ SELECT node_type, result FROM get_citus_tests_label_provider_labels('"user 2"')
...
 DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
-NOTICE:  issuing SELECT worker_create_or_alter_role('user 2', 'CREATE ROLE "user 2" NOSUPERUSER NOCREATEDB NOCREATEROLE INHERIT NOLOGIN NOREPLICATION NOBYPASSRLS CONNECTION LIMIT -1 PASSWORD NULL VALID UNTIL ''infinity''', 'ALTER ROLE "user 2" NOSUPERUSER NOCREATEDB NOCREATEROLE INHERIT NOLOGIN NOREPLICATION NOBYPASSRLS CONNECTION LIMIT -1 PASSWORD NULL VALID UNTIL ''infinity''');SECURITY LABEL FOR "citus '!tests_label_provider" ON ROLE "user 2" IS 'citus ''!unclassified'
+NOTICE:  issuing SELECT worker_create_or_alter_role('user 2', 'CREATE ROLE "user 2" NOSUPERUSER NOCREATEDB NOCREATEROLE INHERIT NOLOGIN NOREPLICATION NOBYPASSRLS CONNECTION LIMIT -1 PASSWORD NULL', 'ALTER ROLE "user 2" NOSUPERUSER NOCREATEDB NOCREATEROLE INHERIT NOLOGIN NOREPLICATION NOBYPASSRLS CONNECTION LIMIT -1 PASSWORD NULL');SECURITY LABEL FOR "citus '!tests_label_provider" ON ROLE "user 2" IS 'citus ''!unclassified'
```
Comment (JelteF) [inline src/test/regress/expected/seclabel.out:172]: Should it maybe be `VALID UNTIL NULL` instead of completely removing it?

Comment (onurctirtir) [reply src/test/regress/expected/seclabel.out:172]: I believe it't not valid to use `NULL` for `valid until`:

```sql
create user new_user password '12345' valid until null;
ERROR:  syntax error at or near "null"
LINE 1: create user new_user password '12345' valid until null;
                                                          ^
```

**Review by gurkanindibay — APPROVED** [review]
Well done :)

File: src/test/regress/expected/seclabel.out (hunk 1 above)
Comment (onurctirtir) [reply src/test/regress/expected/seclabel.out:172]: > I believe it't not valid to use `NULL` for `valid until`:
> 
> ```sql
> create user new_user password '12345' valid until null;
> ERROR:  syntax error at or near "null"
> LINE 1: create user new_user password '12345' valid until null;
>                                                           ^
> ```

What do you think about this @JelteF?

Comment (JelteF) [reply src/test/regress/expected/seclabel.out:172]: Yeah, if that syntax is invalid than obviously not.

**Review by JelteF — APPROVED** [review]

**Pull request merged**